### PR TITLE
chore: minor internal refactors

### PR DESCRIPTION
**Pull request by Marvin9**
As part of university course assignment, I have to perform refactor where possible.

Here is the list of commits and what kind of refactor was followed.

1. https://github.com/wiremock/wiremock/commit/926936bdb67168edab2051c101ecbd9f9f4d3662 [Extract Variable](https://refactoring.guru/extract-variable)
extracted non trivial boolean operations in condition block and put it in self explanatory named variable.

2. https://github.com/wiremock/wiremock/commit/c6f0261ccb7cbf4ba58ae856a9c99c3dd919cc70 [Decompose Conditional](https://refactoring.guru/decompose-conditional)
moved scenario checking condition in `BasicMappingBuilder.build`. Also used `checkState`

3. https://github.com/wiremock/wiremock/commit/58b67239a850408ff611bbeabccc455d668acce4 [Extract Method](https://refactoring.guru/extract-method)
moved header injection and verifying response status in its own methods from `HttpAdminClient.safelyExecuteRequest`

~4. https://github.com/wiremock/wiremock/commit/2c05908116cf4c654eced7c1a349f3fa431f0507 [Extract Class](https://refactoring.guru/extract-class)
I noticed JsonSchemaVersion should not be in `WireMock` class and saw that this was mentioned in the [PR](https://github.com/wiremock/wiremock/pull/2134#discussion_r1165986386). Feel free to enlighten me about the side-effects this change could result or any other suggestion.~

## References

I haven't created an issue but would be happy to create one as I want to implement some more refactors.

## Submitter checklist

- [x] Recommended: Join [WireMock Slack](https://slack.wiremock.org/) to get any help in `#help-contributing` or a project-specific channel like `#wiremock-java`
- [ ] Recommended: If you participate in Hacktoberfest 2023, make sure you're [signed up](https://wiremock.org/events/hacktoberfest/) there and in the WireMock form 
- [x] The PR request is well described and justified, including the body and the references
- [x] The PR title represents the desired changelog entry
- [x] The repository's code style is followed (see the contributing guide)
- [ ] Test coverage that demonstrates that the change works as expected
- [ ] For new features, there's necessary documentation in this pull request or in a subsequent PR to [wiremock.org](https://github.com/wiremock/wiremock.org)

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Hi. Thanks for the pull request. `JsonSchemaVersion` relocation, while reasonable semantically, will break binary compatibility of  items depending on the WireMock Core API. I do not think it worth the effort to migrate, at least until there is a major breaking re-architecture towards WireMock 4

File: src/main/java/com/github/tomakehurst/wiremock/client/WireMock.java [hunk 1]
```diff
@@ -1046,31 +1046,4 @@ public GlobalSettings getGlobalSettings() {
   public static GlobalSettings getSettings() {
     return defaultInstance.get().getGlobalSettings();
   }
-
-  public enum JsonSchemaVersion {
```
[the remark this answers is not in the document]
Comment (Marvin9) [reply src/main/java/com/github/tomakehurst/wiremock/client/WireMock.java:1050]: Totally makes sense. Before I remove that commit, I want to know if there is a possibility for smooth transition, like keep at both places but mark the older one as deprecated?

[the remark this answers is not in the document]
Comment (Marvin9) [reply src/main/java/com/github/tomakehurst/wiremock/client/WireMock.java]: I have reverted changes that could break compatibility.

**Comment by Marvin9** [comment]
@oleg-nenashev Would you mind re-review it?

**Review by oleg-nenashev — APPROVED** [review]
Thanks, these refactorings are fine. They do not reduce the code side or remove duplication, but arguably they improve code readability and might be helpful in the future

**Comment by Marvin9** [comment]
@oleg-nenashev can you please take a look again?

**Review by oleg-nenashev — APPROVED** [review]
Will try to merge it over the weekend

**Pull request merged**